### PR TITLE
[MODORDERS-1195] - Fix karate tests to avoid piece deletion when 1 quantity left

### PR DESCRIPTION
**Pull request by azizbekxm**
## Purpose
https://folio-org.atlassian.net/browse/MODORDERS-1195

## Approach
 - Implement feature for validation
 - Fix other scenarios for this validation
 
 result with all test
image
image

### CODE REVIEW

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    * def headersAdmin = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenAdmin)', 'Accept': 'application/json' }
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenUser)', 'Accept': '*/*' }
+    * configure headers = headersUser
+
+    * callonce variables
+
+    * def fundId = callonce uuid1
+    * def budgetId = callonce uuid2
+
+  Scenario: Create finances
```
Comment (BKadirkhodjaev) [inline acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature:18]: Maybe move this to each scenario as we usually do in our newest tests, also @Positive/@Negative can be added

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    * def piece = $.pieces[0]
+    * def pieceId = piece.id
+
+    # 5. Verify validation for delete Piece with Order Status 'Open'
+    Given path 'orders/pieces', pieceId
+    When method DELETE
+    Then status 422
+    And match $.errors[0].message == "The piece cannot be deleted because it is the last piece for the poLine in with Receiving Workflow 'Synchronized order and receipt quantity' and cost quantity '1'"
+
+    # 6. lose Order
```
Comment (BKadirkhodjaev) [inline acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature:53]: A small typo

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    # 1. Create order
+    * def v = call createOrder { 'id': '#(orderId)' }
+
+    # 2. Create order line with quantity 1 and Receiving Workflow 'Manual'
+    * table poLineLocations
+      | locationId         | quantity | quantityPhysical |
+      | globalLocationsId  | 1        | 1                |
+      | globalLocationsId2 | 1        | 1                |
+    * def v = call createOrderLine { 'id': '#(poLineId)', 'orderId': '#(orderId)', 'quantity': 2, locations: '#(poLineLocations)', 'checkinItems': false }
```
Comment (BKadirkhodjaev) [inline acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature:90]: Nice usage of the tables

**Review by BKadirkhodjaev — APPROVED** [review]

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature (hunk 2 above)
Comment (azizbekxm) [reply acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature]: Thanks, fixed

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    Then status 200
+    And match $.totalRecords == 1
+    * def piece = $.pieces[0]
+    * def pieceId = piece.id
+
+    # 5. Verify validation for delete Piece with Order Status 'Open'
+    Given path 'orders/pieces', pieceId
+    When method DELETE
+    Then status 422
+    And match $.errors[0].message == "The piece cannot be deleted because it is the last piece for the poLine in with Receiving Workflow 'Synchronized order and receipt quantity' and cost quantity '1'"
```
Comment (Saba-Zedginidze-EPAM) [inline acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature:51]: Can we reuse this message and define it once in Background?

**Review by Saba-Zedginidze-EPAM — APPROVED** [review]
LGTM

File: acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature (hunk 4 above)
Comment (azizbekxm) [reply acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature]: Thanks, moved to background

**Pull request merged**